src/DataFixtures/ProjectFeatureFixtures.php

```diff
@@ -17,86 +16,80 @@ class ProjectFeatureFixtures extends Fixture implements DependentFixtureInterfac
 
-        $manager->flush();
+    public function __construct()
+    {
+        $this->faker  = Factory::create('fr_FR');
     }
 
-    private function associateSpecificFeature(ObjectManager $manager)
+    public function load(ObjectManager $manager)
     {
-        $faker  =  Factory::create('fr_FR');
-
-        $categoryCount=count(CategoryFixtures::CATEGORIES);
         $projectCount=count(ProjectFixtures::PROJECTS);
-        $featureMaxIndex=FeatureFixtures::getProjectSpecificFeatureNumber();
+        $specificFeatureCount=FeatureFixtures::getProjectSpecificFeatureNumber();
+        $standardFeatureCount=count(FeatureFixtures::FEATURES);
 
-        $featureCount=0;
+        $projectFeatures=[];
         for ($projectIndex=0; $projectIndex<$projectCount; $projectIndex++) {
-            $project=$this->getReference('project_'.$projectIndex);
+            $project = $this->getReference('project_' . $projectIndex);
 
-            $featureNumber=rand(0, 2 * $featureMaxIndex / $projectCount);//attention aux -1
+            //Associate specific Features
+            $featureNumber=rand(0, 2 * $specificFeatureCount / $projectCount);
             for ($i=0; $i<$featureNumber; $i++) {
-                $projectFeature = new ProjectFeature();
-                $projectFeature->setProject($project);
-                $projectFeature->setDescription($faker->paragraph(self::DESCRIPTION_LENGTH));
-                $projectFeature->setDay(rand(0, 4 * self::MAX_FEATURE_LOAD)/4);
-
-                $categoryIndex=rand(1, $categoryCount-1);
-                $chosenCategory=$this->getReference('category_'.$categoryIndex);
-                $projectFeature->setCategory($chosenCategory);
-
-                if ($featureCount<=$featureMaxIndex) {
-                    $projectFeature->setFeature($this->getReference('specific_feature_'.$featureCount));
+                $specificIndex=count($projectFeatures);
+                if ($specificIndex<$specificFeatureCount) {
+                    $projectFeatures[]=$this->makeSpecificFeature($project, $specificIndex);
                 }
+            }
 
-
-                $manager->persist($projectFeature);
-
-                $this->addReference('specific_project_feature_'.$featureCount, $projectFeature);
-                $featureCount++;
+            //Associate standard Features
+            $featureIndexes=range(0, $standardFeatureCount-1);
+            $featureNumber=rand(self::MIN_STANDARD_FEATURE_PER_PROJECT, $standardFeatureCount-1);
+            for ($i=0; $i<$featureNumber; $i++) {
+                $projectFeatures[]=$this->makeStandardFeature($project, $featureIndexes);
             }
         }
+
+        foreach ($projectFeatures as $feature) {
+            $manager->persist($feature);
+        }
+
+        $manager->flush();
     }
 
-    private function associateStandardFeature(ObjectManager $manager)
+    private function makeSpecificFeature(Project $project, int $specificIndex) : ProjectFeature
```

Comment:
pas simple à suivre, ton code appelle une méthode privée qui appelle une autre méhtode privée... mais dans l'ensemble c'est plutot propre :-)
un conseil pour etre plus clair dans ton code, quand tu fais des choses un peu complexes comme ça, ajoutes un simple bloc au dessus de ta méthode
 /**
* Description en 1 ligne de ce que fais ma méthode (en english of course) 
*/

PHPStorm te permet de générer automatiquement les PHPDocs sur toutes tes méthodes, c'est une bonne pratique quand on commence à bien découper son code ^^

Reply:
fait